sdcm/nemesis.py

```diff
@@ -1117,6 +1117,72 @@ def disrupt_network_start_stop_interface(self):  # pylint: disable=invalid-name
         finally:
             self.target_node.remoter.run("sudo /sbin/ifup eth1")
 
+    def streaming_err_and_rebuild(self, task='decommission'):
+        """
+        Stop streaming task in middle and rebuild the data on the node.
+        """
+        target_node_ip = self.target_node.ip_address
+
+        def streaming_task_thread(nodetool_task='decommission'):
+            """
+            Execute some nodetool command to start persistent streaming
+            task: decommission | rebuild | repair
+            """
+            try:
+                self.target_node.run_nodetool(nodetool_task)
+            except Exception:  # pylint: disable=broad-except
+                self.log.debug('%s is stopped' % nodetool_task)
+
+        log_start = self.target_node.mark_log()
+        streaming_thread = threading.Thread(target=streaming_task_thread, kwargs={'nodetool_task': task})
+        streaming_thread.start()
+
+        self.log.info('interrupt the task (%s) to trigger streaming error' % task)
+        if random.randint(0, 1) == 0:
+            self.log.debug('Interrupt the streaming task by pkill')
+            self.target_node.remoter.run('pkill -9 -f %s' % task)
+        else:
+            self.log.debug('Interrupt the streaming task by hard reboot')
+            self.target_node.reboot(hard=True, verify_ssh=True)
+        streaming_thread.join(60)
+
+        ips = []
+        for dc_info in self.cluster.get_nodetool_status():
+            ips += dc_info.values()
+        if target_node_ip not in ips:
```

Comment:
after rebooting an instance and using public ip connectivity, a target node **public** IP can change.
So need to use `self.target_node.ip_address` instead of saving it in `target_node_ip`

Reply:
another thing, I would make a couple of retries before "declaring" the node as lost and reprovisioning it and reconsidering this fallback mechanism (since we should fallback to such logic on all disruptive nemesis and I am not sure this is the right path to go)
@fruch @roydahan  WUT?

Reply:
> after rebooting an instance and using public ip connectivity, a target node **public** IP can change. So need to use `self.target_node.ip_address` instead of saving it in `target_node_ip`

Thanks, I will fix it.

Reply:
Thanks for the comment, this task isn't high priority now, I will try to address the problem next week.